pkg/ingester/ingester.go

```diff
@@ -1363,7 +1375,26 @@ func (i *Ingester) QueryStream(req *client.QueryRequest, stream client.Ingester_
 		return err
 	}
 
-	i.metrics.queries.Inc()
+	storageValue, matchers, err := removeStorageMatcher(matchers)
+	if err != nil {
+		return err
+	}
+
+	ephemeral := false
+	if storageValue == "" || storageValue == PersistentStorageLabelValue {
+		// storageValue is logged later, so set it to proper value.
+		storageValue = PersistentStorageLabelValue
+	} else if storageValue == EphemeralStorageLabelValue {
+		ephemeral = true
+	} else {
+		return fmt.Errorf(errInvalidStorageLabelValue, storageValue)
+	}
```

Comment:
[nit] To keep this function cleaner, you may consider moving all of this to a dedicated function `detectStorageMatcher()`.

Reply:
I've moved check for accepted label values to `removeStorageMatcherAndGetStorageType` (previously `removeStorageMatcher`) in 
https://github.com/grafana/mimir/pull/3961/commits/2c15e153267c3ba7671204a2d7b16493f00bef1c